sdcclient/_secure.py

```diff
@@ -846,6 +846,19 @@ def delete_compliance_task(self, id):
 
         return True, None
 
+    def run_compliance_task(self, id):
+        '''**Description**
+            Run now the compliance task with the given id
+
+        **Arguments**
+            - id: the id of the compliance task to run
+        '''
+        res = requests.post(self.url + '/api/complianceTasks/run', data=json.dumps({'items': [id]}), headers=self.hdrs, verify=self.ssl_verify)
+        if not self._checkResponse(res):
+            return [False, self.lasterr]
+
+        return [True, res.json()]
```

Comment:
Adding this method might sound logic, but quick comment. We should be returning:
https://github.com/draios/python-sdc-client/blob/master/sdcclient/_common.py#L1010

Reply:
I see in previous pull-requests @davideschiera requesting to don't use this method:
https://github.com/draios/python-sdc-client/pull/89#discussion_r286587649

I'm ok anyway.

Reply:
It sounds like it's time to define one way to go :-)

[`_checkResponse`](https://github.com/draios/python-sdc-client/blob/28a1e913a7cde872b47e2b45925de7d3f7e0a807/sdcclient/_common.py#L29) seems to be used much more often than [`_request_result`](https://github.com/draios/python-sdc-client/blob/28a1e913a7cde872b47e2b45925de7d3f7e0a807/sdcclient/_common.py#L1010). However, `_request_result` essentially encodes a very similar pattern used everywhere with `_checkResponse` (not exactly the same, and unfortunately in some cases data returned for successful requests is not always consistent).

It seems like the most logical way to go would be using `_request_result` everywhere (possibly with an easy way to customize the output in case of successful request). I don't have strong opinions either way though. No matter the decision, I'd suggest to do so in a separate PR dedicated to create consistency in how API responses are processed (FWIW I'd also reconsider the function name, convention aside the intent of "check response" seems to be more clear than "request result").

For what concerns this specific PR, given the highly suboptimal level of consistency we currently have, I wouldn't personally consider the change a blocker.


(PS: I can't recall why I suggested to follow the pattern with `_checkResponse`, I think I got fooled by the usage stats within the scanning file - 25 vs 2 - without thinking much about it. Sorry about the poor analysis).

Reply:
@figarocorso any thoughts on this review?

Reply:
@figarocorso one way or another we're going to get a consistent way to handle API responses. I'll wait few more days for you to chime in before merging the PR. We can always go back and fix it afterwards anyway ;-)